integration-tests/src/main/scala/io/stryker4s/repository/OrderRepository.scala

```diff
@@ -0,0 +1,30 @@
+package io.stryker4s.repository
+
+import cats.Applicative
+import cats.implicits._
+import io.stryker4s.entity.Order
+
+import scala.collection.concurrent.TrieMap
+import scala.language.higherKinds
+
+object OrderRepository {
+  def apply[F[_]: Applicative](): OrderRepository[F] = new OrderRepository[F]()
+}
+
+class OrderRepository[F[_]: Applicative] {
+
+  private val orders = new TrieMap[Int, Order]
+
+  def create(order: Order): F[Order] = {
+    orders += (order.id -> order)
+    order.pure[F]
```

Comment:
Not related to the PR, but how can this be a pure function if it has a side-effect (adding an element to a list)?

Reply:
I think the response is pure in the effect context but the actual writing isn't of course. But this would/could be abstracted using a database.